File: packages/sdk/lib/api/asset.service.ts
import FormData from 'form-data';

import { Asset, AssetInterface } from './asset.interface';
import { Paginated, RequestParameter } from './data.interface';
import { DataService } from './data.service';
import { HttpClient } from './http.service';

export class AssetService extends DataService<Asset> implements AssetInterface {
  constructor(httpClient: HttpClient) {
    super(httpClient, process.env.DEBUG_ASSET_URL || 'api/assets');
  }

  public addAttachment = (id: string, form: FormData): Promise<Asset> => {
    const headers = { ...form.getHeaders() };
    return this.httpClient.post<Asset>(`${this.basePath}/${id}/attachment`, form, {
      headers,
      maxBodyLength: Infinity,
      maxContentLength: Infinity,
    });
  };

  public getChildren(assetId: string, params: RequestParameter = {}): Promise<Paginated<Asset[]>> {
    return this.getManyFiltered({ parent: assetId }, params);
  }
}

File: packages/hahnpro/hpc_api/hpc_api/asset_service.py
from typing.io import IO

from .http_client import HttpClient


class AssetService:
    def __init__(self, http_client: HttpClient):
        self.http_service = http_client
        self.base_path = '/assets'

    # addAttachment
    def get_one(self, id: str, options={}):
        params = {} if ('populate' not in options) else {"populate": options["populate"]}
        return self.http_service.get(f'{self.base_path}/{id}', params).json()

    def add_attachment(self, id: str, file: IO):
        return self.http_service.upload(f'{self.base_path}/{id}/attachment', file).json()
